static/js/containers/ChannelSettingsLink.js

```diff
@@ -15,22 +19,18 @@ export const CHANNEL_SETTINGS_MENU_DROPDOWN = "CHANNEL_SETTINGS_MENU_DROPDOWN"
 
 type Props = {
   channel: Channel,
+  history: Object,
   isOpen: boolean,
   showDropdown: () => void,
-  hideDropdown: () => void
+  hideDropdown: () => void,
+  startFormEdit: () => void
 }
 
 export class ChannelSettingsLink extends React.Component<Props> {
-  renderMenu = () => {
-    const { channel, hideDropdown } = this.props
-
-    return (
-      <DropdownMenu closeMenu={hideDropdown} className="settings-menu-dropdown">
-        <li>
-          <Link to={editChannelBasicURL(channel.name)}>Channel settings</Link>
-        </li>
-      </DropdownMenu>
-    )
+  startEdit = () => {
+    const { channel, history, startFormEdit } = this.props
+    history.push(channelURL(channel.name))
+    startFormEdit()
```

Comment:
It's probably the `clearForm` in the `componentDidMount` of `WidgetListContainer`. I'll see about moving that into the `componentWillUnmount` function to avoid the race condition problem

Reply:
yep, that fixed it!